File: gcpc2019/electionmeddling/submissions/accepted/paul-pq.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int n, m;
	cin >> n >> m;

	vector<int> cost(n);
	for (int i = 0; i < n; i++) {
		int x; cin >> x;

		priority_queue<int> Q;
		for (int j = 0; j < m-1; j++) {
			int y; cin >> y;
			Q.push(y);
		}

		while (x <= Q.top()) {
			x++, cost[i]++;
			Q.push(Q.top()-1);
			Q.pop();
		}
	}

	sort(begin(cost),end(cost));
	cout << accumulate(begin(cost),begin(cost)+(n/2+1),0) << endl;
}

File: gcpc2019/electionmeddling/submissions/accepted/PaulPQ.java
import java.util.*;

public class PaulPQ {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt(), m = sc.nextInt();

		int[] cost = new int[n];
		for (int i = 0; i < n; i++) {
			int x = sc.nextInt();

			PriorityQueue<Integer> Q = new PriorityQueue<>((a,b) -> Integer.compare(b,a));
			for (int j = 0; j < m-1; j++) Q.add(sc.nextInt());

			while (x <= Q.peek()) {
				x++;
				cost[i]++;
				Q.add(Q.poll()-1);
			}
		}
		Arrays.sort(cost);

		int res = 0;
		for (int i = 0; i < n/2+1; i++) res += cost[i];
		System.out.println(res);
	}
}
